tools/perf/ucc_pt_benchmark.cc

```diff
@@ -30,6 +30,8 @@ ucc_pt_benchmark::ucc_pt_benchmark(ucc_pt_benchmark_config cfg,
         if (cfg.op_type != UCC_PT_OP_TYPE_ALLTOALLV) {
             throw std::runtime_error("Only ALLTOALLV is supported for file generator");
         }
+    } else if (cfg.gen.type == UCC_PT_GEN_TYPE_MATRIX) {
+        generator = new ucc_pt_generator_matrix(cfg.gen.matrix.kind, communicator->get_size(), communicator->get_rank(), cfg.dt, cfg.op_type, cfg.gen.nrep, cfg.gen.matrix.token_size_KB_mean, cfg.gen.matrix.num_tokens, cfg.gen.matrix.tgt_group_size_mean);
```

Comment:
add a check that coll type is alltoallv, for other collectives this generator doesn't work